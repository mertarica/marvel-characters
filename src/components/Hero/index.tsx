import React from "react";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import { ICharacter } from "../../types/character.type";

type Props = {
  character: ICharacter;
};

const Hero = ({ character }: Props) => {
  return (
    <Row>
      <Col lg={9} className="order-lg-1 order-2">
        <h1>{character.name}</h1>
        <p>{character.description}</p>
      </Col>
      <Col lg={3} className="order-lg-2 order-1">
        <img
          src={`${character.thumbnail?.path}.${character.thumbnail?.extension}`}
          alt="nothing to list"
          className="w-100"
        />
      </Col>
    </Row>
  );
};

export default Hero;
