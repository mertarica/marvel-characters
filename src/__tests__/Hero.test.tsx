import renderer from "react-test-renderer";
import Hero from "../components/Hero";
import fakeCharacter from "../__mock__/character";

describe("<Hero />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Hero character={fakeCharacter} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
