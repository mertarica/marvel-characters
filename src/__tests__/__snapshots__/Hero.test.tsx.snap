// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-9"
  >
    <h1>
      Fake Hero
    </h1>
    <p>
      Fake hero description
    </p>
  </div>
  <div
    className="col-lg-3"
  >
    <img
      alt="nothing to list"
      className="w-100"
      src="."
    />
  </div>
</div>
`;
